pkg/s3wrapper/filesystem.go

```diff
@@ -0,0 +1,184 @@
+package s3wrapper
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	logutil "github.com/openshift/assisted-service/pkg/log"
+
+	"github.com/moby/moby/pkg/ioutils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type FSClient struct {
+	log     *logrus.Logger
+	basedir string
+}
+
+func NewFSClient(basedir string, logger *logrus.Logger) *FSClient {
+	return &FSClient{log: logger, basedir: basedir}
+}
+
+func (c *FSClient) IsAwsS3() bool {
+	return false
+}
+
+func (f *FSClient) CreateBucket() error {
+	return nil
+}
+
+func (f *FSClient) Upload(ctx context.Context, data []byte, objectName string) error {
+	log := logutil.FromContext(ctx, f.log)
+	filePath := filepath.Join(f.basedir, objectName)
+	if err := os.MkdirAll(path.Dir(filePath), 0755); err != nil {
+		err = errors.Wrapf(err, "Unable to create directory for file data %s", filePath)
+		log.Error(err)
+		return err
+	}
+	if err := ioutil.WriteFile(filePath, data, 0600); err != nil {
+		err = errors.Wrapf(err, "Unable to write data to file %s", filePath)
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (f *FSClient) UploadFile(ctx context.Context, filePath, objectName string) error {
+	log := logutil.FromContext(ctx, f.log)
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		err = errors.Wrapf(err, "Unable to open file %s for upload", filePath)
```

Comment:
just to asking, why need to wrap error if you already log it? not sure who will look on added text

Reply:
So that the reason is returned, in case the caller wants to put it in an event or return it to the user?  Do we usually not do that?

Reply:
not sure we do :)

Reply:
but sounds good idea